### PR TITLE
Tests for upgrade

### PR DESCRIPTION
**Pull request by hpasapp**
- add regression test that:
   - a model saved from 2.3.5 can be imported into the version under test
   - given the same inputs, the model in 2.3.5 gives the same outputs as the version under test
- split tests into 3:
   - unit tests
   - torchscript
   - regression tests

### CODE REVIEW

**Comment by taoleicn** [comment]
@sean-asapp

File: test/test_regression_2.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        dropout=0.1,
+        rescale=False
+    ).eval()
+
+    output_sru_235 = torch.load(args.in_outputs)
+    model.load_state_dict(torch.load(args.in_model))
+    inputs = torch.load(args.in_inputs)
+
+    with torch.no_grad():
+        output_sru_250 = model(inputs)
```
Comment (sean-asapp) [inline test/test_regression_2.py:37]: Probably a huge nitpick, but this doesn't always test version 2.5.0 right? Maybe `output_sru_base` and `output_sru_current` would match your bash vars better?

File: test/test_regression.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+echo CURRENT: ${CURRENT_VERSION}
+echo BASE: ${BASE_VERSION}
+
+git clone -b ${BASE_VERSION} . ../${BASE_VERSION}
+
+(
+    cd ../${BASE_VERSION}
+    virtualenv -p $(which python) .venv
+    source .venv/bin/activate
+    pip install -q torch==1.6.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (sean-asapp) [inline test/test_regression.sh:28]: Is it worth supporting locally running this test? If so I think you'd need to detect the OS and drop the `+cpu` package suffix for Mac users, if not then you might add Linux under "assumptions" above 🤷‍♀️

**Comment by sean-asapp** [comment]
This is awesome!!! 🎉 🎉

File: test/regression/test_regression.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# This tests for backward compatibility to BASE_VERSION of sru:
```
Comment (sean-asapp) [inline test/regression/test_regression.sh:3]: Is there any way we can test this in CCIE?

Comment (hpasapp) [reply test/regression/test_regression.sh]: forgot to reomte this file. removed in 1d3b1d7

Comment (hpasapp) [reply test/regression/test_regression.sh]: (so the regression test is now just a standard unit test, which is nice)

Comment (hpasapp) [reply test/regression/test_regression.sh]: (and it's super fast too..)

**Review by taoleicn — APPROVED** [review]

**Review by sean-asapp — APPROVED** [review]

**Review by iitzco-asapp — APPROVED** [review]
LGTM.

At some point, lets move to tox to manage versioning support.

**Comment by taoleicn** [comment]
:tada:

**Pull request merged**